docs/TEST_MIGRATION_COMPLETE.md

```diff
@@ -0,0 +1,377 @@
+# SQLx Test Migration Complete
```

Comment:
I think this is cruft that we don't want committed to the repo. 

There's some good bits in here that are worth capturing in perpetuity, but a lot of it is only relevant to this PR, and is either already captured in the commit messages, or can go in the PR description.